website/static/js/projectSettings.js

```diff
@@ -8,13 +8,19 @@ var $osf = require('js/osfHelpers');
 var oop = require('js/oop');
 var ChangeMessageMixin = require('js/changeMessage');
 
-var NodeCategorySettings = oop.extend(
+var NodeCategoryTitleDescriptionSettings = oop.extend(
     ChangeMessageMixin,
     {
         constructor: function(category, categories, updateUrl, disabled) {
             this.super.constructor.call(this);
 
             var self = this;
+            self.currentNode = window.contextVars.node;
```

Comment:
Re-usable modules should not depend on global variables because globals make testing and reusability difficult. Instead, pass the information you need to the constructor.

When you do this, you might want to refactor the constructor to take a params object rather than positional arguments, e.g.

```js
new ProjectSettings({category: 'project', currentTitle: ...., currentDescription: ...});
```

The only place global variables should be accessed are in the page modules.